### PR TITLE
IGNITE-12211: Client connections system view

### PR DESCRIPTION
**Pull request by nizhikov**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java [hunk 1]
```diff
@@ -62,6 +69,9 @@
 import static org.apache.ignite.internal.processors.metric.GridMetricManager.GC_CPU_LOAD_DESCRIPTION;
 import static org.apache.ignite.internal.processors.metric.GridMetricManager.SYS_METRICS;
 import static org.apache.ignite.internal.processors.metric.impl.MetricUtils.metricName;
+import static org.apache.ignite.internal.processors.odbc.ClientListenerProcessor.CLI_CONN_SYS_VIEW;
+import static org.apache.ignite.internal.processors.odbc.jdbc.JdbcConnectionContext.CURRENT_VER;
+import static org.apache.ignite.internal.processors.platform.client.ClientConnectionContext.DEFAULT_VER;
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: The client from the same source will have same version, isn't it?

File: modules/core/src/main/java/org/apache/ignite/spi/systemview/view/ClientConnectionView.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        return ctx.connectionId();
+    }
+
+    /** @return Connection type. */
+    public String type() {
+        if (ctx == null)
+            return null;
+
+        switch (ctx.type()) {
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/spi/systemview/view/ClientConnectionView.java]: I just want to reuse existing code here.
No need to introduce new enum with the same values as byte constants.
What do you think?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java [hunk 3]
```diff
@@ -243,6 +249,29 @@ public void testServices() throws Exception {
...
+            host = ignite.configuration().getLocalHost();
+
+        int port = ignite.configuration().getClientConnectorConfiguration().getPort();
+
+        try (IgniteClient client =
+                 Ignition.startClient(new ClientConfiguration().setAddresses(host + ":" + port))) {
+
+            try (Connection conn =
+                     new IgniteJdbcThinDriver().connect("jdbc:ignite:thin://" + host, new Properties())) {
+
```
Comment (alex-plekhanov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java]: Redundant NL

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java [hunk 4]
```diff
@@ -243,6 +249,29 @@ public void testServices() throws Exception {
...
+        String host = ignite.configuration().getClientConnectorConfiguration().getHost();
+
+        if (host == null)
+            host = ignite.configuration().getLocalHost();
+
+        int port = ignite.configuration().getClientConnectorConfiguration().getPort();
+
+        try (IgniteClient client =
+                 Ignition.startClient(new ClientConfiguration().setAddresses(host + ":" + port))) {
+
```
Comment (alex-plekhanov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/metric/SqlViewExporterSpiTest.java]: Redundant NL

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java [hunk 5]
```diff
@@ -62,6 +70,9 @@
...
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: Version still hasn't been fixed.

File: modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-plekhanov) [reply modules/core/src/test/java/org/apache/ignite/internal/metric/JmxExporterSpiTest.java]: No, the client from the same source can have another version

**Pull request merged**